compiler-core/src/parse.rs

```diff
@@ -1213,6 +1214,15 @@ where
                             &mut estack,
                             &do_reduce_clause_guard,
                         );
+                    } else if t == Token::LeftParen {
```

Comment:
So this would give the same error even if it wasn't actually a function call but just an open parentheses?
```gleam
match foo {
  _ if hello( -> todo
//          ^ CallInCaseGuard
}
```
That doesn't seem ideal

Reply:
Well I think practically this is what mostly corresponds to the user, what can it be other than a call? If no, what do you suggest?

Reply:
It should have the previous behaviour in that case